### ISSUE TITLE
False positive for API definition in web worker

### ISSUE THREAD

**Issue by cybaea** [obsidianmd/eslint-plugin#150]
I experience a false positive source code warning in the community site scorecard tests. I get

- Warning: Use 'activeDocument' instead of 'document' for popout window compatibility.
    - [src/workers/indexer.worker.ts:103](https://github.com/cybaea/obsidian-vault-intelligence/blob/eed44583d06ca22c8d220a636c9076138a2a8343/src/workers/indexer.worker.ts#L103)

This is not a popup window but an interface definition for [Orama](https://oramasearch.com/) and I can't rename the external API. (Also, this is a web worker which doesn't have a `document` or `activeDocument`)

I am adding a `// eslint-disable-next-line obsidianmd/prefer-active-doc` directive with comments, but it feels like this particular usage of a class attribute called `document` should not trigger the positive.

```typescript
interface OramaResultHit {
    // eslint-disable-next-line obsidianmd/prefer-active-doc -- Orama search results use the 'document' property name; this is a data property in a Web Worker, not a global DOM access.
    document: OramaDocument;
    id: string;
    score: number;
}
```

---

PS: Thanks for the community.obsidian.md site - it's awesome!

**Comment by cybaea** [comment]
See also #162
